extensions/amp-selector/1.0/selector.js

```diff
@@ -239,23 +241,31 @@ function SelectorWithRef(
   );
 
   return (
-    <Comp
-      {...rest}
-      role={role}
-      aria-disabled={disabled}
-      aria-multiselectable={multiple}
-      disabled={disabled}
-      keyboardSelectMode={keyboardSelectMode}
-      multiple={multiple}
-      onKeyDown={onKeyDown}
-      tabIndex={
-        tabIndex ?? keyboardSelectMode === KEYBOARD_SELECT_MODE.SELECT ? 0 : -1
-      }
-    >
-      <SelectorContext.Provider value={context}>
-        {children}
-      </SelectorContext.Provider>
-    </Comp>
+    <>
+      <Comp
+        {...rest}
+        role={role}
+        aria-disabled={disabled}
+        aria-multiselectable={multiple}
+        disabled={disabled}
+        form={form}
+        keyboardSelectMode={keyboardSelectMode}
+        multiple={multiple}
+        name={name}
+        onKeyDown={onKeyDown}
+        tabIndex={
+          tabIndex ?? keyboardSelectMode === KEYBOARD_SELECT_MODE.SELECT
+            ? 0
+            : -1
+        }
+        value={selected}
+      >
+        <SelectorContext.Provider value={context}>
+          {children}
+        </SelectorContext.Provider>
+      </Comp>
+      <input hidden defaultValue={selected} name={name} form={form} />
```

Comment:
Why `defaultValue` and not `value`? If IIRC the `defaultValue` is set only once and here we want it to update every time, right?

Reply:
I don't think it should matter which we use in this case, since controlled/uncontrolled inputs is more a pattern for when a component controls an input that a user can also type into. `defaultValue` updates the `input` value every time `selected` changes here. 

Tried both of them and noticed that `defaultValue` and `value` both allow user to remove `hidden` attribute in the DOM and type into the input, so the only meaningful difference I can tell is that `defaultValue` prop sets the `value` attribute in the DOM node and `value` prop does not. In both cases reading/setting the `element.value` property yields what you would expect.

Reply:
Sounds good.